src/main/java/gregtech/api/model/ModelFactory.java

```diff
@@ -0,0 +1,147 @@
+
+/**
+ * Revamped from https://github.com/LoliKingdom/Zairyou/blob/main/src/main/java/zone/rong/zairyou/api/client/Bakery.java
+ */
+public class ModelFactory {
+
+    private static final Map<ItemCameraTransforms.TransformType, TRSRTransformation> blockTransformationMap = new EnumMap<>(ItemCameraTransforms.TransformType.class);
+    private static final Map<ItemCameraTransforms.TransformType, TRSRTransformation> itemTransformationMap = new EnumMap<>(ItemCameraTransforms.TransformType.class);
+
+    private static FaceBakery INSTANCE;
+
+    static {
+        blockTransformationMap.put(ItemCameraTransforms.TransformType.GUI, getTransform(0, 0, 0, 30, 225, 0, 0.625f));
+        blockTransformationMap.put(ItemCameraTransforms.TransformType.GROUND, getTransform(0, 2, 0, 0, 0, 0, 0.25f));
+        blockTransformationMap.put(ItemCameraTransforms.TransformType.FIRST_PERSON_RIGHT_HAND, getTransform(0, 0, 0, 0, 45, 0, 0.4f));
+        blockTransformationMap.put(ItemCameraTransforms.TransformType.THIRD_PERSON_RIGHT_HAND, getTransform(0, 0, 0, 0, 0, 0, 0.4f));
+        blockTransformationMap.put(ItemCameraTransforms.TransformType.FIRST_PERSON_LEFT_HAND, getTransform(0, 0, 0, 45, 0, 0, 0.4f));
+        blockTransformationMap.put(ItemCameraTransforms.TransformType.THIRD_PERSON_LEFT_HAND, getTransform(0, 0, 0, 45, 0, 0, 0.4f));
+
+        itemTransformationMap.put(ItemCameraTransforms.TransformType.GUI, getTransform(0, 0, 0, 0, 0, 0, 1f));
+        itemTransformationMap.put(ItemCameraTransforms.TransformType.GROUND, getTransform(0, 2, 0, 0, 0, 0, 0.5f));
+        itemTransformationMap.put(ItemCameraTransforms.TransformType.FIRST_PERSON_RIGHT_HAND, getTransform(1.13f, 3.2f, 1.13f, 0, -90, 25, 0.68f));
+        itemTransformationMap.put(ItemCameraTransforms.TransformType.THIRD_PERSON_RIGHT_HAND, getTransform(0, 3, 1, 0, 0, 0, 0.55f));
+        itemTransformationMap.put(ItemCameraTransforms.TransformType.FIRST_PERSON_LEFT_HAND, getTransform(1.13f, 3.2f, 1.13f, 0, 90, -25, 0.68f));
+        itemTransformationMap.put(ItemCameraTransforms.TransformType.THIRD_PERSON_LEFT_HAND, getTransform(0f, 4.0f, 0.5f, 0, 90, -55, 0.85f));
+    }
+
+    public static FaceBakery getBakery() {
+        if (INSTANCE == null) {
+            INSTANCE = new FaceBakery();
+        }
+        return INSTANCE;
+    }
+
+    public static TRSRTransformation getBlockTransform(ItemCameraTransforms.TransformType transformType) {
+        return blockTransformationMap.get(transformType);
+    }
+
+    public static TRSRTransformation getItemTransform(ItemCameraTransforms.TransformType transformType) {
+        return itemTransformationMap.get(transformType);
+    }
+
+    private static TRSRTransformation getTransform(float tx, float ty, float tz, float ax, float ay, float az, float s) {
+        return new TRSRTransformation(new Vector3f(tx / 16, ty / 16, tz / 16), TRSRTransformation.quatFromXYZDegrees(new Vector3f(ax, ay, az)), new Vector3f(s, s, s), null);
+    }
```

Comment:
I believe there is something similar to this in the CCL API, @Yefancy you remember where they reside? That will eliminate the 2 enum maps above this code too.

Reply:
Seems to be in `codechicken.lib.util.TransformUtils`, though it varies from what we have *slightly*. Might be all we need though, I'm not sure